test/jdk/jdk/internal/util/VersionTest.java

```diff
@@ -0,0 +1,108 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+
+import java.util.stream.Stream;
+
+import jdk.internal.util.Version;
+
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * @test
+ * @summary test jdk.internal.util.Version
+ * @modules java.base/jdk.internal.util
+ * @run junit VersionTest
+ */
+
+public class VersionTest {
+
+    private static Stream<Arguments> versionParams() {
+        return Stream.of(
+                Arguments.of("1", new Version(1, 0)),
+                Arguments.of("1.2", new Version(1, 2)),
+                Arguments.of("1.2", new Version(1, 2, 0)),
+                Arguments.of("1.2.3", new Version(1, 2, 3)),
+                Arguments.of("1-abc", new Version(1, 0, 0)), // Ignore extra
+                Arguments.of("1.2-abc", new Version(1, 2, 0)), // Ignore extra
+                Arguments.of("1.2.3.4", new Version(1, 2, 3)), // Ignore extra
+                Arguments.of("1.2.3-abc", new Version(1, 2, 3)) // Ignore extra
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("versionParams")
+    public void checkParse(String verName, Version expected) {
+        Version actual = Version.parse(verName);
+        assertEquals(actual, expected, "Parsed version mismatch");
+    }
+
+    private static Stream<Arguments> illegalVersionParams() {
+        return Stream.of(
+                Arguments.of("1.", IllegalArgumentException.class),     // dot without digits
+                Arguments.of("1.2.", IllegalArgumentException.class),   // dot without digits
+                Arguments.of("1.-abc", IllegalArgumentException.class), // dot without digits
+                Arguments.of("1.2.-abc", IllegalArgumentException.class), // dot without digits
+                Arguments.of("", IllegalArgumentException.class),
+                Arguments.of("xaaa", IllegalArgumentException.class),
+                Arguments.of("abc.xyz", IllegalArgumentException.class)
+        );
+    }
+
+    @ParameterizedTest()
+    @MethodSource("illegalVersionParams")
+    public void checkIllegalParse(String verName, Class<Throwable> exception) {
+        try {
+            Version.parse(verName);
+        } catch (Throwable th) {
+            System.err.printf("%s: %s%n", verName, th.getMessage());
+        }
+        assertThrows(exception, () -> Version.parse(verName));
+    }
```

Comment:
The explicit creation of argument instances can be avoided by using something like:

```java
    @ParameterizedTest
    @ValueSource(strings = {
            // dot without digits
            "1.", "1.2.", "1.-abc", "1.2.-abc",
            // empty
            "",
            // no an initial digit
            "xaaa", "abc.xyz"
    })
    void checkIllegalParse(String verName) {
        checkIllegalParse(verName, IllegalArgumentException.class);
    }

    void checkIllegalParse(String verName, Class<? extends Throwable> exception) {
        assertThrows(exception, () -> Version.parse(verName));
    }
```